components/core/src/clp/ffi/KeyValuePairLogEvent.hpp

```diff
@@ -0,0 +1,81 @@
+#ifndef CLP_FFI_KEYVALUEPAIRLOGEVENT_HPP
+#define CLP_FFI_KEYVALUEPAIRLOGEVENT_HPP
+
+#include <memory>
+#include <optional>
+#include <unordered_map>
+#include <utility>
+
+#include <outcome/single-header/outcome.hpp>
+
+#include "../time_types.hpp"
+#include "SchemaTree.hpp"
+#include "SchemaTreeNode.hpp"
+#include "Value.hpp"
+
+namespace clp::ffi {
+/**
+ * Class for key-value pair log events. Each key-value pair log event contains the following items:
+ *  - A list of key-value pairs
+ *  - A reference to the schema tree
+ *  - The UTC offset of the current log event
+ */
+class KeyValuePairLogEvent {
+public:
+    // Types
+    using KeyValuePairs = std::unordered_map<SchemaTreeNode::id_t, std::optional<Value>>;
```

Comment:
How about `NodeIdValuePairs`? I think `KeyValuePairLogEvent` is an appropriate name since it represents the abstraction, but for the more concrete types and variables, I think it's clearer to name them more directly.

Similarly, I would rename the `kv_pair` variables to `node_id_value_pairs`.